src/libs/ReportUtils.ts

```diff
@@ -3305,9 +3304,12 @@ function getModifiedExpenseOriginalMessage(
         originalMessage.oldCurrency = TransactionUtils.getCurrency(oldTransaction);
         originalMessage.oldMerchant = TransactionUtils.getMerchant(oldTransaction);
 
-        originalMessage.amount = TransactionUtils.getAmount(updatedTransaction, isFromExpenseReport);
-        originalMessage.currency = TransactionUtils.getCurrency(updatedTransaction);
-        originalMessage.merchant = TransactionUtils.getMerchant(updatedTransaction);
+        const modifiedData = TransactionUtils.calculateAmountForUpdatedWaypointOrRate(oldTransaction, transactionChanges, policy, isFromExpenseReport);
```

Comment:
NAB: Maybe call this `modifiedDistanceFields` or something similarly descriptive?